code/game/machinery/medal_printer.dm

```diff
@@ -0,0 +1,85 @@
+/obj/machinery/r_n_d/fabricator/mechanic_fab/autolathe/medal_printer
+	name = "medal printer"
+	desc = "Prints a variety of medal accessories."
+	icon = 'icons/obj/machines/mechanic.dmi'
+	icon_state = "medalprinter"
+	icon_state_open = "medalprinter_t"
+	nano_file = "medalprinter.tmpl"
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK | EMAGGABLE
+	research_flags = NANOTOUCH | TAKESMATIN | HASOUTPUT | IGNORE_CHEMS | HASMAT_OVER
+	var/locked = 1
```

Comment:
this whole locked var is weird, why don't you just put a req_access = (access_hop) on it and check allowed() in the attackby (for labeling)? it's already handled for using the interface at a fabricator level

Reply:
removed and sorted